AI/VCAI/Goals.cpp

```diff
@@ -1219,11 +1219,58 @@ bool CollectRes::fulfillsMe(TSubgoal goal)
 	return false;
 }
 
+int GatherTroops::getCreaturesCount(const CArmedInstance * army)

```

Comment:
Is this function supposed to be used to determine value of army to gather? If yes, use creature AI Value in all relevant places so 3 pikemen won't be treated as better than 2 angels etc.

Reply:
No, it is used to check if there is anything available at all. GATHER_TROOPS is for quests and it is bound to gather exact amount of something not army of certain power. For the last we have GatherArmy